src/bridge/cockpitfsread.c

```diff
@@ -238,6 +249,19 @@ cockpit_fsread_prepare (CockpitChannel *channel)
       return;
     }
 
+  if (fstat (self->fd, &statbuf) < 0)
+    {
+      cockpit_channel_fail (channel, "internal-error",
+                            "%s: couldn't stat: %s", self->path, strerror (errno));
+      return;
+    }
+
+  if (statbuf.st_size > max_read_size)
+    {
```

Comment:
Thanks, looks good! My remaining nitpick is that `!= S_IFREG` should be error code `"not-supported"` instead of `"internal-error"` (stat failure is fine as `internal-error`, of course). But I leave that up to you if you want to change it again.